provider/consul/consul_catalog_config_test.go

```diff
@@ -0,0 +1,1253 @@
+		{
+			desc: "Should return for.bar key value random when empty prefix",
+			tags: []string{
+				"foo.bar=ramdom",
+				"backend.wei=42",
+			},
+			key:          "foo.bar",
+			defaultValue: "random",
+			expected:     "ramdom",
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			p := &CatalogProvider{
+				Domain: "localhost",
+				Prefix: test.prefix,
+			}
+
+			actual := p.getAttribute(test.key, test.tags, test.defaultValue)
+			assert.Equal(t, test.expected, actual)
+		})
+	}
+}
+
+func TestCatalogProviderGetIntAttribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue int
+		expected     int
+	}{
+		{
+			desc:     "should return default value when empty name",
+			name:     "",
+			tags:     []string{"traefik.foo=10"},
+			expected: 0,
+		},
+		{
+			desc:     "should return default value when empty tags",
+			name:     "traefik.foo",
+			tags:     nil,
+			expected: 0,
+		},
+		{
+			desc:     "should return default value when value is not a int",
+			name:     "foo",
+			tags:     []string{"traefik.foo=bar"},
+			expected: 0,
+		},
+		{
+			desc:     "should return a value when tag exist",
+			name:     "foo",
+			tags:     []string{"traefik.foo=10"},
+			expected: 10,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			result := p.getIntAttribute(test.name, test.tags, test.defaultValue)
+
+			assert.Equal(t, test.expected, result)
+		})
+	}
+}
+
+func TestCatalogProviderGetInt64Attribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue int64
+		expected     int64
+	}{
+		{
+			desc:     "should return default value when empty name",
+			name:     "",
+			tags:     []string{"traefik.foo=10"},
+			expected: 0,
+		},
+		{
+			desc:     "should return default value when empty tags",
+			name:     "traefik.foo",
+			tags:     nil,
+			expected: 0,
+		},
+		{
+			desc:     "should return default value when value is not a int",
+			name:     "foo",
+			tags:     []string{"traefik.foo=bar"},
+			expected: 0,
+		},
+		{
+			desc:     "should return a value when tag exist",
+			name:     "foo",
+			tags:     []string{"traefik.foo=10"},
+			expected: 10,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			result := p.getInt64Attribute(test.name, test.tags, test.defaultValue)
+
+			assert.Equal(t, test.expected, result)
+		})
+	}
+}
+
+func TestCatalogProviderGetBoolAttribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue bool
```

Comment:
`defaultValue` is not used in test cases

Reply:
fixed